internal/clients/clients_interfaces.go

```diff
@@ -30,19 +29,12 @@ type ImageBuilder interface {
 	Ready(ctx context.Context) error
 }
 
-var GetEC2Client func(ctx context.Context) (EC2, error)
-var GetEC2ClientWithRegion func(ctx context.Context, region string) (EC2, error)
+var GetAssumedEC2Client func(ctx context.Context, arn string) (EC2, error)
+var GetAssumedEC2ClientWithRegion func(ctx context.Context, arn string, region string) (EC2, error)
```

Comment:
One of these functions can be simply defined in the interface package as it will always be the same, implementations need only to redefine one of them:

```go
func GetAssumedEC2Client func(ctx context.Context, arn string) (EC2, error) {
  return GetAssumedEC2ClientWithRegion(ctx, arn, config.AWS.Region)
}
```

I know it was me who sort of pushed @adiabramovitch into this, with factory functions it do look little bit weird. I am good with this (definitely better than variadic function), but one solution to this could be simply having just a single function and when region is set to empty string, then config default would be used. 

The reason why I am bringing it here is that @avitova will soon need also non-assumed ec2 client for her instance type preloading work which can be done directly on our service account. This could possibly double the amount of factory functions.

I also suggest to clean naming a bit, to be consistent across clouds I propose the following terms:

* GetServiceClient - for our service (direct) account
* GetCustomerClient - for the customer's (assumed) client